---
title: "Food Borne Pathogens"
author: "Anthony Girard, Timothy Hedspeth, Yutong Li"
date: '2022-10-05'
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```


# Introduction
## Motivating question 
Have you ever woken up in the morning and saw a news headline, like "E Coli. Outbreak at Chipolte leads to national recall"? Though it is easy to shrug this off and move on, but in the United States we note that every year there are an [estimated 37 million cases of food borne illness](https://www.cdc.gov/foodborneburden/pdfs/scallan-estimated-illnesses-foodborne-pathogens.pdf).  

## Literature review 


# Data Overview 

## What data are we using 
The data that we working with in this analysis is an isolate of Samonella, which is utilized so we understand the data 


```{r}

#~~~~~~~~~~~~#
## Packages ## 
#~~~~~~~~~~~~#

library(lubridate) # working with dates 
library(tidyverse) # working with data 

# Read in the data 
setwd("~/Desktop/Semester_3/Practical/Final")
salmonella <- read.csv("salmonella_2013_2022.csv")
ecoli <- read.csv("E.coli_2013-2022.csv")
campylobacter <- read.csv("Campylobacter_2013-2022.csv")




apply(isolated, 2, typeof)
# All the variables are character 

#levels(as.factor(isolated$X.Organism.group))
# Given that this 


# All of the variables are charcater 
isolated$Create.date <- as.Date(isolated$Create.date)
barplot(table(month(isolated$Create.date)))




```


In our analysis we have a longitudinal data set with data regarding salmonella, ranging from 2013 to 2022. 



```{r}

```








```{r}
sum(table(isolated$Strain) > 1)
sum(table(isolated$Strain) == 1)

#levels(as.factor(isolated$Strain))
which(table(isolated$Strain) > 1)
```

There are 16,246 rows of the data frame that correspond to a single strain of the virus, these strains likely coulld be excluded they are rare, and there are 3 strains that are commonly observed, AG3, Enteritdis, Salmonella enterica subsp. enterica. For the sake of future analysis it would likely be beneficial to find these common strains and group by them to get common strains of the virus. 

```{r}
isolated$col_num <- seq(1:nrow(isolated)) # Make a column number so we can keep
                                          # track of columns of interest

sum(table(isolated$Isolate.identifiers) > 1) # This seems to be a unique variable
                                             # to each observation, likely not 
                                             # relevant for our analysis

#isolates_identifiers <- isolated %>% separate_rows(Isolate.identifiers, sep=",\\s*")
```





```{r}
sum(table(isolated$Isolate) > 1) # Unique to each observation, not revelant 
```



```{r}
sum(table(isolated$Create.date) > 1) # We can look at date info, this will 
                                     # probably be the best 
```

```{r}
sum(table(isolated$Location) >1) # 87 locations 
```





```{r}
table(isolated$Isolation.type)
```


```{r}
table(isolated$Min.same)
```


```{r}
table(isolated$Min.diff)
```


```{r}
sum(table(isolated$BioSample) > 1) # This probably isn't super relevant 
```


```{r}
sum(table(isolated$Assembly) > 1)
which(table(isolated$Assembly) > 1)
```

```{r}
sum(table(isolated$Computed.types) >1 )
which(table(isolated$Computed.types) >1)
```


## Variables of interest 

For our analysis there are a few variables that are of interest to our analysis. The first of which is the **Collection.date** variable, indicating when the sample was collected. **AMR_genotypes** is the Antimicrobial resistance genes, where a missing value indicates that there is no antimicrobial  resistance detected. **AST_phenotypes** tells us the phenotype based on testing, **host** is the host species, **host_disease** could be missing, **isolate_identifiers** a list alternative isolates, **isolation_source** describes the physical, environmental and/or local geographical source of the biological sample from which the sample was derived, if provided by the submitter, **epi_type** can be either clinical or environmental, **k-mer** is the organism work, **Min-same** is a measure of closeness to other isolates of the same type, **outbreak** if there is an unusal number of cases in an area, **erd_group** is the SNP cluster, **source_type** where the data comes from. 



## Missing Data 

```{r}

#~~~~~~~~~~~~~~~~#
## Missing Data ## 
#~~~~~~~~~~~~~~~~#

# The first objective of this EDA is to explore the missing patterns in our 
# data at the surface level 


# Replace Missing strings with NA 
salmonella[salmonella == ""] <- NA 
ecoli[ecoli == ""] <- NA
campylobacter[campylobacter == ""] <- NA 


## Missingness by column

# Salmonella
na_by_cols_sal <- as.data.frame(apply(salmonella,2, is.na))
sum_na_by_col_sal <- apply(na_by_cols_sal,2,sum)/nrow(na_by_cols_sal)
sum_na_by_col_sal <- as.data.frame(round(sum_na_by_col_sal,4))

# E. Coli 
na_by_cols_ecoli <- as.data.frame(apply(ecoli,2, is.na))
sum_na_by_col_ecoli <- apply(na_by_cols_ecoli,2,sum)/nrow(na_by_cols_ecoli)
sum_na_by_col_ecoli <- as.data.frame(round(sum_na_by_col_ecoli,4))

# Camplyobaceter 
na_by_cols_camp <- as.data.frame(apply(campylobacter,2, is.na))
sum_na_by_col_camp <- apply(na_by_cols_camp,2,sum)/nrow(na_by_cols_camp)
sum_na_by_col_camp <- as.data.frame(round(sum_na_by_col_camp,4))

# Get the data together 
Missing_by_col <- cbind(sum_na_by_col_sal, sum_na_by_col_ecoli, sum_na_by_col_camp)
names(Missing_by_col) <- c("x", "y", "z") # Create a dummy variable for filtering 
Missing_by_col <- Missing_by_col %>% filter(x != 0 & 
                                            y != 0 & 
                                            z != 0   )
names(Missing_by_col) <- c("Salmonella", "E. Coli", "Campylobacter")

# Print the data frame 
Missing_by_col %>% kbl(caption = "Variables with some missing observations")



## Missingness by row

# Using the commented out code below we check to see if any of the data is 
# complete for a given observation 

#sum(complete.cases(salmonella))
#sum(complete.cases(ecoli))
#sum(complete.cases(campylobacter))

# There are no complete cases in the data set 

# Salmonella 
na_by_row_sal <- as.data.frame(apply(salmonella,1, is.na))
sum_na_by_row_sal <- apply(na_by_row_sal,2,sum)/nrow(na_by_row_sal)
salmonella_missing <- as.data.frame(sum_na_by_row_sal)

# E. Coli 
na_by_row_ecoli <- as.data.frame(apply(ecoli,1, is.na))
sum_na_by_row_ecoli <- apply(na_by_row_ecoli,2,sum)/nrow(na_by_row_ecoli)
ecoli_missing <- as.data.frame(sum_na_by_row_ecoli)

# Campylobacter
na_by_row_camp <- as.data.frame(apply(campylobacter,1, is.na))
sum_na_by_row_camp <- apply(na_by_row_camp,2,sum)/nrow(na_by_row_camp)
campylobacter_missing <- as.data.frame(sum_na_by_row_camp)



ggplot(data = salmonella_missing, aes(x = sum_na_by_row_sal, color = "sum_na_by_row_sal")) + 
  geom_density(alpha = .9) + 
  geom_density(data = ecoli_missing, aes(x = sum_na_by_row_ecoli,
                                         color = "sum_na_by_row_ecoli"), alpha = .9) + 
  geom_density(data = campylobacter_missing, aes(x = sum_na_by_row_camp, 
                                                 color = "sum_na_by_row_camp"), alpha = .9)
  

```

From the information we can see that there is a great deal of missingness in this data set and the peracent missingness is relatley high for most of the variables. Though there is heterogeneity in the data, with some variables being observed frequently with one data set and being missing in the data set. 








## Exploratory Analysis 
We note that the data in this study is longitudinal in nature, we sought to quantify the trends in 
```{r, message=FALSE}
# Let us consider looking at trends over time

# We must have our data in a form that is workable form for analysis 
salmonella$Create.date <- as.Date(salmonella$Create.date) 
ecoli$Create.date <- as.Date(ecoli$Create.date)
campylobacter$Create.date <- as.Date(campylobacter$Create.date)

# Decompose the dates to day month year

# Day
salmonella$day <- day(salmonella$Create.date)
ecoli$day <- day(ecoli$Create.date)
campylobacter$day <- day(campylobacter$Create.date)

# Month 
salmonella$month <- month(salmonella$Create.date)
ecoli$month <- month(ecoli$Create.date)
campylobacter$month <- month(campylobacter$Create.date)

# Year 
salmonella$year <- year(salmonella$Create.date)
ecoli$year <- year(ecoli$Create.date)
campylobacter$year <- year(campylobacter$Create.date)




# How many Samonella cases per year? 
Num_per_year_salmonella <- salmonella %>% 
                           group_by(year) %>% 
                           summarize(num = n())

# How many ecoli cases per year? 
Num_per_year_ecoli <- ecoli %>% 
                      group_by(year) %>% 
                      summarize(num = n())

# How many ecoli cases per year? 
Num_per_year_campylobacter <- campylobacter %>% 
                              group_by(year) %>% 
                              summarize(num = n())

# Lets look at the trend over time, and 
ggplot(data = Num_per_year_salmonella, aes(x = year, y = num, color = "Salmonella")) +
  geom_point() + 
  geom_line() +
  # E coli 
  geom_point(data = Num_per_year_ecoli, aes(x = year, y = num, color = "E coli")) + 
  geom_line(data = Num_per_year_ecoli, aes(x = year, y = num, color = "E coli")) + 
  # campylobacter
  geom_point(data = Num_per_year_campylobacter, aes(x = year, y = num, color = "Campylobacter")) + 
  geom_line(data = Num_per_year_campylobacter, aes(x = year, y = num, color = "Campylobacter")) + 
  theme_minimal() + 
  labs(title= "Number of Samonella cases by year", 
       x = "Year", 
       y = "Number of cases")





# Number of cases per month 
salmonella$month1 <- month(salmonella$Create.date)
salmonella <- salmonella %>% mutate(month = case_when(month1 == 1 ~ "Jan", 
                                                      month1 == 2 ~  "Feb", 
                                                      month1 == 3 ~ "Mar", 
                                                      month1 == 4 ~ "Apr", 
                                                      month1 == 5 ~ "May", 
                                                      month1 == 6 ~ "Jun", 
                                                      month1 == 7 ~ "Jul", 
                                                      month1 == 8 ~ "Aug", 
                                                      month1 == 9 ~ "Sep", 
                                                      month1 == 10 ~ "Oct", 
                                                      month1 == 11 ~ "Nov", 
                                                      month1 == 12 ~ "Dec"))

Num_per_month_sal <- salmonella %>%  
                     group_by(year, month) %>% 
                     summarize(num =  n())
Num_per_month_sal$month <- factor(Num_per_month_sal$month, levels = c("Jan", "Feb", 
                                                                    "Mar", "Apr", "May", 
                                                                    "Jun", "Jul", "Aug", 
                                                                    "Sep", "Oct", 
                                                                    "Nov", "Dec"))
ggplot(data = Num_per_month_sal, aes(x = as.numeric(month), y = num)) + 
  geom_point(aes(color = as.factor(year))) + 
  scale_x_continuous(breaks=seq(1,12,1), labels=c("Jan", "Feb", 
                                                  "Mar", "Apr", "May", 
                                                  "Jun", "Jul", "Aug", 
                                                  "Sep", "Oct", 
                                                  "Nov", "Dec"))+
  geom_smooth()
  

table(isolated$Isolation.source)

table(isolated[c(grep("animal", isolated$Isolation.source),grep("Animal", isolated$Isolation.source))
               ,]$Isolation.source)
```

We need to begin by doing some cleaning of the data that we have in our data file, one of the main variables that we are interested in is the isolation source, as tracing the source of the infection is of great interest for our analysis. In its original form this is a factor variable with over 6,300 levels, which is due to poor reporting practices, the first objective is to greatly reduce the heterogeneity of this variable to make the analysis achievable. We will start by looking at identifying potential sources that may be reported frequently in many forms such as chicken and their by products like eggs. This analysis requires the assumption that these observations will map to true chicken data which may be a strong assumption, but without this the analysis would not be plausible.  


```{r}
# This will be the cleaning of the data regarding the isolation source
# Given the extreme heterogeniety in 

# We note that there are going to be some sources 

length(levels(as.factor(salmonella$Isolation.source)))
table(salmonella$Isolation.type)

# Lets start with chicken
salmonella[grep("chicken", salmonella$Isolation.source),9] <- "Chicken"
salmonella[grep("Chicken", salmonella$Isolation.source),9] <- "Chicken"
salmonella[grep("egg", salmonella$Isolation.source),9] <- "Chicken"
salmonella[grep("Egg", salmonella$Isolation.source),9] <- "Chicken"
salmonella[grep("Poultry", salmonella$Isolation.source),9] <- "Chicken"
salmonella[grep("poultry", salmonella$Isolation.source),9] <- "Chicken"



# length(levels(as.factor(salmonella$Isolation.source))) # This reduces by a factor of approx 500 

# Lets do beef 
salmonella[grep("beef", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("Beef", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("cow", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("Cow", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("Cattle", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("cattle", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("Veal", salmonella$Isolation.source),9] <- "Cow/Beef"
salmonella[grep("veal", salmonella$Isolation.source),9] <- "Cow/Beef"


length(levels(as.factor(salmonella$Isolation.source)))


# Lettuce 
salmonella[grep("Lettuce", salmonella$Isolation.source),9] <- "Lettuce"
salmonella[grep("lettuce", salmonella$Isolation.source),9] <- "Lettuce"

length(levels(as.factor(salmonella$Isolation.source)))

# Milk 
salmonella[grep("Milk", salmonella$Isolation.source),9] <- "Milk"
salmonella[grep("milk", salmonella$Isolation.source),9] <- "Milk"

length(levels(as.factor(salmonella$Isolation.source)))

# Almonds 



salmonella[grep("Animal Feed", salmonella$Isolation.source),9] <- "Animal feed"

# Raw/ food borne stuff 
#salmonella[grep("Raw", salmonella$Isolation.source), 9] <- "Raw"
#salmonella[grep("raw", salmonella$Isolation.source), 9] <- "raw"

#salmonella[grep()]


# abscess 
salmonella[grep("abscess", salmonella$Isolation.source), 9] <- "abscess"
salmonella[grep("Abscess", salmonella$Isolation.source), 9] <- "abscess"

# chicken 



table(salmonella$Isolation.source)
```



```{r}
sum(is.na(salmonella$Min.same))/nrow(salmonella)
```



# Conclusion 


